Data/HashMap/Array.hs

```diff
@@ -346,6 +368,26 @@ updateWith' :: Array e -> Int -> (e -> e) -> Array e
 updateWith' ary idx f = update ary idx $! f (index ary idx)
 {-# INLINE updateWith' #-}
 
+-- | This newtype wrapper is to avoid confusion when local functions
+-- take more than one paramenter of 'Int' type (see 'go' in
+-- 'Data.HashMap.Base.unionWithKeyInternal').
+newtype Size = Size { unSize :: Int }
+    deriving (Eq, Ord, Num, Integral, Enum, Real)
```

Comment:
I introduced this was a few months ago because I got a nasty bug caused by mixing up the bitmask argument in most of the `go` helpers with the then-existing size accumulator (that has since been removed in favor of the `Sized` approach).

I don't mind making a separate PR to add these where possible/sensible.

Reply:
I wrote totally bogus functions for `Data.IntMap` in a similar way. Then I goofed adding them to the test suite so the problem wasn't caught and I had to deprecate releases. Embarrassing.